taichi/backends/metal/taichi_file_util.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "taichi/backends/metal/kernel_util.h"
+
+TLANG_NAMESPACE_BEGIN
+
+class Kernel;
+class SNode;
+
+namespace metal {
+struct BufferSize {
+    int64_t root_buffer_size;
```

Comment:
Ditto for the remaining

```suggestion
    int64_t root_buffer_size{0};
```